indigo_app/static/javascript/indigo/views/document_editor.js

```diff
@@ -115,12 +115,13 @@
 
               pasteTable(table);
             }
```

Comment:
I think we need a `return` here, just after the for loop. Then, the `else` statement on line 120 can fall away. We should execute 121-124 in two situations: there's HTML but not tables, or there's no HTML.